mantidproject/mantid
# Exclude Range during Fit in Muon interfaces

**Description of work.**
This PR adds the ability to exclude a range when doing a fit in the Muon Analysis and FDA interfaces.

**To test:**
1. Open the Muon Analysis interface
2. Load MUSR62260
3. Go to the Fitting tab
4. Add a GausOsc function and set Frequency parameter to 1.0
4. Tick `Exclude Range` in the fitting options
5. Change the `Exclude Start X` to 2.0
6. Click `Fit`, and it should be poor. Click `Undo Fit`
7. Now untick the `Exclude Range`, click `Fit` and the provided fit is better.

This shows that a range is indeed being excluded for the first fit, causing a poor fit (in this case).

Fixes #32018

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
